components/snap_recovery/src/services.rs

```diff
@@ -136,7 +142,7 @@ impl<ER: RaftEngine> RecoveryService<ER> {
         // config rocksdb l0 to optimize the restore
         // also for massive data applied during the restore, it easy to reach the write
         // stop
-        let db = engines.kv.clone();
+        let db: &RocksEngine = engines.kv.get_disk_engine();
```

Comment:
I think it's better to use kv directly and internally it passes the call to disk_engine. 
The principle is that keep minimal code change outside of the engine components.